lib/EnsEMBL/REST/Model/ga4gh/variantannotations.pm

```diff
@@ -0,0 +1,541 @@
+
+  my $sift_analysis;
+  $sift_analysis->{result} = $tva->sift_prediction();
+  $sift_analysis->{score}  = $tva->sift_score();
+ 
+  if (defined $sift_analysis->{result}){
+    ## TODO: define better id!
+    $sift_analysis->{analysisId}    = 'SIFT.5.2.2';
+    push @analysisResults , $sift_analysis;
+  }    
+      
+  my $polyphen_analysis;
+  $polyphen_analysis->{result} = $tva->polyphen_prediction();
+  $polyphen_analysis->{score}  = $tva->polyphen_score();
+
+  if (defined $polyphen_analysis->{result}){
+    ## TODO: define better id!
+    $polyphen_analysis->{analysisId}     = 'Polyphen.2.2.2_r405';
+    push @analysisResults , $polyphen_analysis;
+  }
+
+  return \@analysisResults ;
+}
+
+
+## put required features ids/ SO accessions in a hash for look up
+
+sub extract_required{
+
+  my $self     = shift;
+  my $req_list = shift;
+  my $type     = shift;
+  my $req_hash;
+
+  foreach my $required ( @{$req_list} ){
+    $req_hash->{$required->{term}} = 1 if $type eq 'SO';
+  }
+
+  return $req_hash;
+}
+
+## create temp feature set name from current db version
+## replace with GA4GH id when format available
+
+sub get_version{
+
+  my $self = shift;
+  my $data = shift;
+
+  ## extract required meta data from variation database to create set name
+  my $core_ad = $self->context->model('Registry')->get_DBAdaptor('homo_sapiens', 'core' );
+  my $meta_ext_sth = $core_ad->dbc->db_handle->prepare(qq[ select meta_key, meta_value from meta]);
+  $meta_ext_sth->execute();
+  my $stuff = $meta_ext_sth->fetchall_arrayref();
+
+  my %meta;
+  foreach my $l(@{$stuff}){
+    $meta{$l->[0]} = $l->[1] if defined $l->[1];
+  }
+
+  ## need better ids
+  $data->{current_set}      = 'Ensembl.' . $meta{schema_version} . '.'. $meta{"assembly.default"};
+  $data->{current_version}  = $meta{schema_version};
+  return $data;
+}
+
+sub searchVariantAnnotations_compliance{
+
+  my $self = shift;
+  my $data = shift;
+
+  my $variantSetId = (split/\:/,  $data->{variantAnnotationSetId})[1];
+
+  ##VCF collection object for the compliance annotation set
+  $data->{vcf_collection} =  $self->context->model('ga4gh::ga4gh_utils')->fetch_VCFcollection_by_id( $variantSetId );
+  Catalyst::Exception->throw(" Failed to find the specified variantAnnotationSetId")
+    unless defined $data->{vcf_collection}; 
+
+  ## look up filename from vcf collection object
+  my $file  =  $data->{vcf_collection}->filename_template();
+
+  $file =~ s/\#\#\#CHR\#\#\#/$data->{referenceName}/;
+
+  # return these ordered by position for simple pagination
+  my @var_ann;
+  my $nextPageToken;
+  ## VCF is 1-based; GA4GH is 0-based & open-closed 
+  $data->{end}++;
+  $data->{pageToken} = $data->{start} unless defined $data->{pageToken}  && $data->{pageToken}  =~/\d+/;
+
+  ## exits here if unsupported chromosome requested
+  return (\@var_ann, $nextPageToken) unless -e $file;
+ 
+  my $parser = Bio::EnsEMBL::IO::Parser::VCF4Tabix->open( $file ) || die "Failed to get parser : $!\n";
```

Comment:
This kind of initialisation should probably not be left until user request, unless you've got millions of file handles in play. Consider initialising as a Moose property with a builder sub{}